lib/smile-identity-core/id_api.rb

```diff
@@ -17,12 +17,10 @@ def initialize(partner_id, api_key, sid_server)
       end
     end
 
-    def submit_job(partner_params, id_info)
+    def submit_job(partner_params, id_info, options = {})
       self.partner_params = symbolize_keys partner_params
-
-      @timestamp = Time.now.to_i
-
       self.id_info = symbolize_keys id_info
+      @use_legacy_sec_key = symbolize_keys(options || {}).fetch(:use_legacy_sec_key, true)
```

Comment:
why would you need `options || {}` if your are assigning options as an empty hash when not provided on line 20?

Reply:
the option is `signature`. We unfortunately can't default to it, because then it breaks people's callbacks or existing implementations because `sec_key` isn't there. Why is the option called `signature`? because that's what the initial implementation was and we want the libraries to all look the same, and have things named identically.

Reply:
Okay, that makes sense. I should've checked the other library work first. FWIW, it does currently default to using the sec_key, as we discussed.

Reply:
> why would you need options || {} if your are assigning options as an empty hash when not provided on line 20?

Because it can still be passed as nil; setting it in the signature only determines what happens when it's not provided.

Reply:
@beaglebets I changed `use_legacy_sec_key` to `signature` in 3fcdff8.